source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,166 @@
+.. _csharp-serialization:
+
+=============
+Serialization
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about serialization with the {+driver-long+}.
+Serialization is the process of mapping a C# object to and from a BSON document.
+
+Serializer Registry
+-------------------
+
+The serializer registry is an implementation of ``IBsonSerializerRegistry``
+that contains all registered serializers. You can access the serializer registry
+with the ``SerializerRegistry`` property of the ``BsonSerializer``
+class as follows:
+
+.. code-block:: csharp
+
+   var intSerializer = BsonSerializer.SerializerRegistry.GetSerializer<int>();
+
+.. important::
+
+   The serializer registry is a global registry. This means that you cannot use
+   multiple registries in a single application.
+
+Serialization Provider
+----------------------
+
+A serialization provider allows you to provide serializers to your
+serialization registry. The serialization provider should be registered as early
+as possible in your application to ensure that the serializers are used by the
+registry. Your serialization provider must implement the
+``IBsonSerializationProvider`` interface. 
+
+Example
+~~~~~~~
+
+The following example implements a serialization provider. The provider has a
+``GetSerializer`` method that returns a serializer based on the type of the
+input. If the method input is an ``int`` it returns a custom ``Int32``
+serializer, otherwise it returns ``null`` so that it can be handled by another
+provider. 
+
+.. code-block:: csharp
+   
+   class MyProvider : IBsonSerializationProvider
+   {
+       public IBsonSerializer GetSerializer(Type type)
+       {
+           if (type == typeof(int))
+           {
+               return new MyInt32Serializer();
```

Comment:
Q: I'm not sure how complex these custom serializer classes are. Would it be helpful to show how to define one on this page?

Reply:
Removed serialization provider section